File: Server/src/main/java/biz/kakee/pvo/events/request/common/DeviceInfo.java
package biz.kakee.pvo.events.request.common;

import lombok.Getter;
import lombok.ToString;

@ToString
public class DeviceInfo {
    @Getter
    private String uniqueID;

    @Getter
    private String instanceID;

    @Getter
    private String deviceID;

    @Getter
    private String manufacturer;

    @Getter
    private String model;

    @Getter
    private String brand;

    @Getter
    private String systemName;

    @Getter
    private String systemVersion;

    @Getter
    private String bundleID;

    @Getter
    private String buildNumber;

    @Getter
    private String version;

    @Getter
    private String readableVersion;

    @Getter
    private String deviceName;

    @Getter
    private String userAgent;

    @Getter
    private String locale;

    @Getter
    private int tzOffsetHours;

    @Getter
    private String country;
}

File: Mobile/Shared/jsx/DeviceInfo.js
'use strict';

import DeviceInfo from 'react-native-device-info';

class UserDeviceInfo {
    info() {
        if (this._info) {
            return this._info;
        }

        this._info = {
            uniqueID: DeviceInfo.getUniqueID(),
            instanceID: DeviceInfo.getInstanceID(),
            deviceID: DeviceInfo.getDeviceId(),
            manufacturer: DeviceInfo.getManufacturer(),
            model: DeviceInfo.getModel(),
            brand: DeviceInfo.getBrand(),
            systemName: DeviceInfo.getSystemName(),
            systemVersion: DeviceInfo.getSystemVersion(),
            bundleID: DeviceInfo.getBundleId(),
            buildNumber: DeviceInfo.getBuildNumber(),
            version: DeviceInfo.getVersion(),
            readableVersion: DeviceInfo.getReadableVersion(),
            deviceName: DeviceInfo.getDeviceName(),
            userAgent: DeviceInfo.getUserAgent(),
            locale: DeviceInfo.getDeviceLocale(),
            tzOffsetHours: new Date().getTimezoneOffset() / 60,
            country: DeviceInfo.getDeviceCountry()
        };

        return this._info;
    }
}

export default new UserDeviceInfo();
